File: 2025-12/src/day_12/prms_42746/Solution.java
package day_12.prms_42746;

import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String[] arr = new String[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            arr[i] = String.valueOf(numbers[i]);
        }
        Arrays.sort(arr, (o1, o2) -> (o2 + o1).compareTo(o1 + o2));

        String ans = String.join("", arr);

        if (ans.charAt(0) == '0') {
            return "0";
        } else {
            return ans;
        }
    }
}

File: 2025-12/src/day_12/prms_42746/Solution.cpp
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

string solution(vector<int> numbers) {
    vector<string> v;
    for (int n : numbers) v.push_back(to_string(n));
    sort(v.begin(), v.end(), [](const string& a, const string& b) {
        return b + a < a + b;
    });

    string ans;
    for (string& s : v) {
        ans += s;
    }

    if (ans[0] == '0') {
        return "0";
    } else {
        return ans;
    }
}
